spec/rdiff-backup-test/create_server_spec.rb

```diff
@@ -0,0 +1,174 @@
+require_relative '../spec_helper'
+
+describe 'rdiff-backup-test::create_server' do
+  [CENTOS_6_OPTS, CENTOS_7_OPTS].each do |pltfrm|
+    context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
+      let(:runner) do
+        ChefSpec::SoloRunner.new(
+          pltfrm.dup.merge(step_into: ['rdiff-backup'])
+        )
+      end
+      cached(:chef_run) { runner.converge(described_recipe) }
+      before do
+        allow(Chef::EncryptedDataBagItem).to receive(:load).with(
+          'rdiff-backup-secrets', 'secrets'
+        ).and_return(
+          key: 'secret-key'
+        )
+      end
+
+      it do
+        expect(chef_run).to create_rdiff_backup('test1')
+      end
+      # Tests lines [74:end] of the libraries/rdiff-backup.rb file
+      it do
+        expect(chef_run).to include_recipe('yum-epel')
+      end
+      # line 76
+      %w(rdiff-backup cronolog).each do |p|
+        it do
+          expect(chef_run).to install_package(p)
+        end
+      end
+      # lines 77:97, because new_resource.nrpe defaults to true
+      it do
+        expect(chef_run).to include_recipe('nrpe')
+      end
+      it do
+        expect(chef_run).to create_cookbook_file(
+          chef_run.node['nrpe']['plugin_dir'] + '/check_rdiff'
+        ).with(
+          mode: 0755,
+          owner: chef_run.node['nrpe']['user'],
+          group: chef_run.node['nrpe']['group'],
+          source: 'nagios/plugins/check_rdiff',
+          cookbook: 'rdiff-backup'
+        )
+      end
+      it do
+        expect(chef_run).to add_nrpe_check('check_rdiff_job_test1').with(
+          command: '/usr/bin/sudo /usr/lib64/nagios/plugins/check_rdiff' \
+            ' -w 16 '\
+            '-c 18 '\
+            '-r /you/are/my/only/hope '\
+            '-p 24 '\
+            '-l 800000000'
+        )
+      end
+      # line 100
+      it do
+        expect(chef_run).to create_user('rdiff-backup-server')
+      end
+      # line 101
+      it do
+        # group == owner, so group is NOT created here
+        expect(chef_run).to_not create_group('rdiff-backup-server')
+      end
+      # lines 102:108
+      it do
+        expect(chef_run).to install_sudo('rdiff-backup-server').with(
+          user: 'rdiff-backup-server',
+          group: 'rdiff-backup-server',
+          nopasswd: true,
+          commands: ['/usr/bin/sudo rdiff-backup '\
+                     '--server --restrict-read-only /']
+        )
+      end
+      # lines 109:114
+      it do
+        expect(chef_run).to create_directory('/var/rdiff-backup/locks').with(
+          owner: 'rdiff-backup-server',
+          group: 'rdiff-backup-server' || 'rdiff-backup-server',
```

Comment:
Is this conditional really needed here? We should be expecting one value given the current given parameters.